lab_13/src/lib.c

```diff
@@ -0,0 +1,250 @@
+#include "lib.h"
+
+node_t *create_node(vehicle_t data)
+{
+    node_t *node = (node_t *)malloc(sizeof(node_t));
+    node->data = data;
+    node->next = NULL;
+    node->prev = NULL;
+    return node;
+}
+
+void push_front(forward_list_t *list, vehicle_t data)
+{
+    node_t *node = create_node(data);
+    node->next = list->head->next;
+    if (list->head->next != NULL)
+    {
+        list->head->next->prev = node;
+    }
+    list->head->next = node;
+    node->prev = list->head;
+
+    if (list->tail == list->head)
+    {
+        list->tail = node;
+    }
+}
+
+void pop_front(forward_list_t *list)
+{
+    if (list->head->next == NULL)
+    {
+        return;
+    }
+    node_t *for_deletion = list->head->next;
+    list->head->next = for_deletion->next;
+    if (for_deletion->next != NULL)
+    {
+        for_deletion->next->prev = list->head;
+    }
+    else
+    {
+        list->tail = list->head;
+    }
+    free(for_deletion);
+}
+
+void display_list(forward_list_t *list)
+{
+    node_t *current = list->head->next;
+    while (current != NULL)
+    {
+        printf("Brand: %s, Model: %s, Year: %d, Owner: %s, ID: %s\n",
+               current->data.brand,
+               current->data.model,
+               current->data.year,
+               current->data.owner.name,
+               current->data.owner.id);
+        current = current->next;
+    }
+    printf("NULL\n");
+}
+
+void display_list_2(node_t *head)
+{
+    node_t *current = head;
+    while (current != NULL)
+    {
+        printf("Brand: %s, Model: %s, Year: %d, Owner: %s, ID: %s\n",
+               current->data.brand,
+               current->data.model,
+               current->data.year,
+               current->data.owner.name,
+               current->data.owner.id);
+        current = current->next;
+    }
+    printf("NULL\n");
+}
+
+void display_list_id(forward_list_t *list)
```

Comment:
What is the reason to have `display_list1` and `display_list2`?

Reply:
Roudiment from base code, will clean this